File: trunk/OscarDotNet/component/biz/Class_biz_user.pas
unit Class_biz_user;

interface

uses
  Class_db_user,
  Class_db_users,
  ki,
  system.web;

type
  TClass_biz_user = class
  private
    db_user: TClass_db_user;
    db_users: TClass_db_users;
  public
    constructor Create;
    function EmailAddress: string;
    function IdNum: string;
    function Privileges: ki.string_array;
    function Roles: ki.string_array;
  end;

implementation

constructor TClass_biz_user.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
  db_user := TClass_db_user.Create;
  db_users := TClass_db_users.Create;
end;

function TClass_biz_user.EmailAddress: string;
begin
  EmailAddress := db_users.PasswordResetEmailAddressOfId(IdNum);
end;

function TClass_biz_user.IdNum: string;
begin
  IdNum := db_users.IdOf(httpcontext.current.user.identity.name);
end;

function TClass_biz_user.Privileges: ki.string_array;
begin
  Privileges := db_users.PrivilegesOf(IdNum);
end;

function TClass_biz_user.Roles: ki.string_array;
begin
  Roles := db_user.RolesOf(IdNum);
end;

end.

File: trunk/OscarDotNet/component/biz/Class_biz_user.cs
using System;

using System.Web;
using Class_db_user;
using Class_db_users;
namespace Class_biz_user
{
    public class TClass_biz_user
    {
        private TClass_db_user db_user = null;
        private TClass_db_users db_users = null;
        //Constructor  Create()
        public TClass_biz_user() : base()
        {
            // TODO: Add any constructor code here
            db_user = new TClass_db_user();
            db_users = new TClass_db_users();
        }
        public string EmailAddress()
        {
            string result;
            result = db_users.PasswordResetEmailAddressOfId(IdNum());
            return result;
        }

        public string IdNum()
        {
            string result;
            result = db_users.IdOf(HttpContext.Current.User.Identity.Name);
            return result;
        }

        public string[] Privileges()
        {
            string[] result;
            result = db_users.PrivilegesOf(IdNum());
            return result;
        }

        public string[] Roles()
          {
          var role_array = db_user.RolesOf(IdNum());
          if (role_array.Length == 0)
            {
            role_array = (new string[] {"Member"});
            }
          return role_array;
          }

    } // end TClass_biz_user

}
